### ISSUE TITLE
Run `wasm-strip` with `--release`

### ISSUE THREAD

**Issue by surma** [rustwasm/wasm-pack#543]
## 💡 Feature description
Building the default example project as a release build yields some rather big wasm modules, which might be deterring to some:

Command: `wasm-pack build --release`
- stable: 1.3M
- nightly: 771K

After running `wasm-strip`, the size goes down drastically:
- stable: 1.6K
- nightly: 1.6K

Anything speaking against running `wasm-strip` automatically, at least for `--release`?

wasm-pack 0.6.0
rustc 1.32.0 (9fda7c223 2019-01-16)
rustc 1.34.0-nightly (57d7cfc3c 2019-02-11)

**Comment by ashleygwilliams** [comment]
@surma yeah! i was actually thinking about this after our in person hack sesh at the meetup. the biggest blocker to adding tools to wasm-pack is the availability of a rust library and/or linux/macOS/windows binaries (the binaries are the preferred option). would you be willing to take a look into the availability of these binaries? if they exist, adding wasm-strip should be otherwise not too difficult. 

also a side note- wasm-pack defaults to release so this is a feature that would be run by default and turned off for when wasm-pack runs in --dev mode.

**Comment by fitzgen** [comment]
I think we would only want to run `wasm-strip` for the `release` profile, not `debug` or `profiling` profiles, since it will remove debug info. But I'm also a little confused because I would assume that we are already not including any custom sections in `release` profile builds...

Either way -- if we add support for this, we would want to control it via the toml config.

**Comment by ashleygwilliams** [comment]
@fitzgen agree we'd only want it for --release, which is the "default" (i.e. not on --dev, or debug, or the others)

it certainly appears that `wasm-strip` does something here, perhaps it would help to figure out exactly what it does! (re: your comment re custom sections)

**Comment by surma** [comment]
As I mentioned in the title, I definitely meant `wasm-strip` to only be run for release builds!

I ran Twiggy real quick:
<img width="818" alt="screen shot 2019-02-13 at 8 38 02 pm" src="https://user-images.githubusercontent.com/234957/52743834-d84be180-2fd2-11e9-8ac3-76fff8be760c.png">

So it seems that custom sections do not get stripped.

(That being said, I think this might have been a change in 1.32 or something. I _believe_ when I was on 1.30 the binary was way smaller)

**Comment by fitzgen** [comment]
Figured out why the DWARF isn't getting stripped: https://github.com/rustwasm/wasm-bindgen/issues/1254

**Comment by ashleygwilliams** [comment]
@surma could you try the newest release of wasm-bindgen and confirm this is fixed? (i'd expect that `wasm-strip` would make a lot less of, if no, difference if the bug is indeed fixed!)

**Comment by surma** [comment]
I updated `wasm-bindgen` dependency to `0.2.37`, but sadly it doesn’t seem to actually be invoked.

Invoking `wasm-strip` manually shaves off around 3k.

**Comment by ashleygwilliams** [comment]
@fitzgen circling back up on this - any idea what wasm-strip is stripping? @surma can you give any more details?

**Comment by fitzgen** [comment]
`wasm-bindgen` doesn't ever attempt to run `wasm-strip` -- I think there is some kind of misunderstanding here?

I think we need to be more concrete about examples, expectations, and actual behavior here, so it would be useful to have:

* steps to reproduce (clone this repo, run this command, etc)
* expected behavior (the generated wasm does not have any custom sections)
* actual behavior (the generated wasm has X and Y custom sections)

**Comment by ashleygwilliams** [comment]
@surma sorry for the Extreme Delay. any status update on this? if you have a way to repro that'd be great.

**Comment by surma** [comment]
Apologies! I lost track of this as well!

Here’s a repro repo: https://github.com/surma-dump/wasm-pack-issue-543

Repro steps:
1. Run `wasm-pack build --release`
2. Check file size of `pkg/wasm_pack_issue_543_bg.wasm`
3. Run `wasm-strip pkg/wasm_pack_issue_543_bg.wasm`
4. Check file size of `pkg/wasm_pack_issue_543_bg.wasm` again

Expected: File size remains the same

Currently observed: File size got smaller after running `wasm-strip`

**Comment by wdanilo** [comment]
I've got a different story to share:
1. Clone https://github.com/luna/basegl/commit/9d1b18d6816310c46d7ea3aa2f507792976563e7
2. Build with `./scripts/build.sh` (wrapper for wasm-pack).
3. Check file size of `target/web/basegl_bg.wasm`
4. Run `wasm-strip` on the file.
5. Check the file size again.

Result: **After running `wasm-strip` the size grows from 3.9Mb to 4.1Mb**.
